content/concepts/did-ddo.md

```diff
@@ -34,98 +43,120 @@ In Ocean, a DID is a string that looks like:
-    "attributes": {
-      "main": {},
-      "additionalInformation": {},
-      "curation": {}
-    }
-  },
-  {
-    "index": 1,
-    "type": "access",
-    "serviceEndpoint": "http://localhost:8030/api/v1/provider/services/consume",
-    "attributes": {
-      "main": {
-        "cost":"10",
-        "timeout":0
-      },
-      "additionalInformation": {}
-    }
-  },
-  {
-    "index": 2,
-    "type": "compute",
-    "serviceEndpoint": "http://localhost:8030/api/v1/provider/services/compute",
-    "attributes": {
-      "main": {
-        "cost":"10",
-        "timeout":3600
-      },
-      "additionalInformation": {}
-    }
-  }
-]
+
+## Service
+
+| Attribute           | Type                  | Required | Description                                                                                                                                                                                       |
+| ------------------- | --------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **`type`**     | Text          | **Yes**       | Type of service (access, compute, wss, etc |
+| **`name`**     | Text          | No       | Service friendly name |
+| **`serviceEndpoint`**     | Text          | **Yes**       | Provider URI |
+| **`timeout`**     | Number          | **Yes**       | describing how long the sevice can be used after consumption is initiated. A timeout of 0 represents no time limit. |
+| **`files`**         | Array of files object |**No **   | Array of `File` objects including the encrypted file urls that overwrites the root files object for this service [Files](#files)   |
+
+
+Depending on the service type, the following attributes are applied:
+
+### Compute datasets attributes
+
+An asset with a service of type `compute` has the following additional attributes under `privacy` object :
+
+| Attribute                    |   Type                |   Required  | Description                                               |
+| ---------------------------- | ----------------------| ----------- |---------------------------------------------------------- |
+| **`allowRawAlgorithm`**      | `boolean`             | yes         | If True, a drag & drop algo can be runned                 |
+| **`allowNetworkAccess`**     | `boolean`             | yes         | If True, the algo job will have network access (stil WIP) |
+| **`publisherTrustedAlgorithmPublishers `**      | Array of `String`     | yes         | If Empty , then any published algo is allowed. Otherwise, only published algorithms by some publishers are allowed |
+| **`publisherTrustedAlgorithms `**      | Array of `Objects`     | yes         | If Empty , then any published algo is allowed. (see below) |
```

Comment:
should be array of type `TrustedAlgorithms` where the type is already defined below